src/constraint_system/composer.rs

```diff
@@ -54,11 +50,13 @@ use std::collections::BTreeMap;
 /// Each gate or group of gates adds an specific functionallity or operation to
 /// the circuit description, and so, that's why we can understand
 /// the StandardComposer as a builder.
-#[derive(Debug)]
-pub struct StandardComposer<
+#[derive(derivative::Derivative)]
+#[derivative(Debug)]
```

Comment:
`derivative::Derivative` makes the traits easier to derive so that they have correct generics semantics. The default Rust `derive` macro will always put the trait you are trying to derive on all the generics instead of the actual fields. It's a (soft) bug in the original design, and now it's too late to change it. If you want to replicate what `derivative::Derivative` does, you would need to implement all the traits yourself. You don't need this for objects which do not have generics, but if they do, then most likely you need it or you'll have the wrong semantics.

Example:
```rust
struct Point<E, P>
where
    E: PairingField,
    P: TEModelParameters<BaseField = E::Fr>,
{
    x: Variable,
    y: Variable,
    __: PhantomData<(E, P)>,
}
```

with `derive(Debug)`, the trait requirements would be `E: Debug, P: Debug` which is not correct. If you use `derivative(Debug)` it would have not extra requirements. In this case, you don't need to specify bounds because `PhantomData` is handled properly, but in other cases, you need to specify bounds manually.

`arkworks` also uses this sometimes, it's just very convenient when you have generics.